etcdctl/ctlv3/command/snapshot_command.go

```diff
@@ -409,7 +409,7 @@ func dbStatus(p string) dbstatus {
 
 	ds := dbstatus{}
 
-	db, err := bolt.Open(p, 0400, nil)
+	db, err := bolt.Open(p, 0400, &bolt.Options{ReadOnly: true})
```

Comment:
we still need the nofsync option. also the file is already opened as readonly (0400), it is strange that we write it somehow...

can you also add an e2e test for the case that you described to prevent any potential regression?

Reply:
oh. i looked at the open code in boltdb, it does not try to interpret the fileperm bit anyway...

Reply:
only ReadOnly is OK; bbolt will detect if there's a free list or not, which is why it was rebuilding the free list here

Reply:
it seems there is no test case for etcdctl ... should i add a new test file for snapshot_command or somewhere else ?

Reply:
@zrss 

see https://github.com/coreos/etcd/blob/master/e2e/ctl_v3_snapshot_test.go

Reply:
@xiang90 PTAL